### PR TITLE
Service dialogs sorting

### PR DESCRIPTION
**Pull request by chalettu**
@miq-bot add_label bug
BZ https://bugzilla.redhat.com/show_bug.cgi?id=1519358

### CODE REVIEW

**Comment by chalettu** [comment]
@eclarizio , would you mind taking a look and reviewing?

File: src/dialog-user/services/dialogData.spec.ts [hunk 1]
```diff
@@ -124,5 +122,16 @@ describe('DataTableSettingsService test', () => {
...
+      values: [
+        [0, 'B'],
+        [5, 'C'],
+        [2, 'A']
+      ],
+      options: { sort_by: 'description', sort_order: 'descending' }
+    };
+    const testSortedDescription = dialogData.updateFieldSortOrder(testDropDownDescription);
+    const expectedSortedResult = [[5, 'C'], [0, 'B'], [2, 'A']];
+    expect(testSortedDescription).toEqual(expectedSortedResult);
```
Comment (eclarizio) [inline src/dialog-user/services/dialogData.spec.ts]: I think I'd rather see this as a separate `it` block, but I don't care that much it's not a necessary change, just wanted to point it out.

File: src/dialog-user/services/dialogData.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ export default class DialogDataService {
       field.fieldValidation = '';
       field.errorMessage = '';
     }
-    if (field.type === 'DialogFieldDropDownList') {
+    const sortableFieldTypes = ['DialogFieldDropDownList', 'DialogFieldRadioButton'];
+    if (sortableFieldTypes.indexOf(field.type) > -1) {
```
Comment (eclarizio) [inline src/dialog-user/services/dialogData.ts]: Can we use lodash's `_.includes` here?

**Comment by ohadlevy** [comment]
@chalettu whats the status on this one? thanks :)

**Comment by chalettu** [comment]
@ohadlevy , I am updating the PR today and retesting.  Should have it up here soon.

File: src/dialog-user/services/dialogData.spec.ts (hunk 1 above)
Comment (chalettu) [reply src/dialog-user/services/dialogData.spec.ts]: Yeah, sometimes I like to group tests together.  I think it should be fine.

**Comment by himdel** [comment]
@miq-bot add_label gaprindashvili/backported
@miq-bot remove_label gaprindashvili/yes

This was released in manageiq-ui-components 1.0.4, backported to gaprindashvili via ManageIQ/manageiq-ui-classic#3175 .

**Pull request merged**